### PR TITLE
grcov: init at 0.8.7

### PR DESCRIPTION
**Pull request by DieracDelta**
###### Motivation for this change

Add [grcov](https://github.com/mozilla/grcov) to nixpkgs.

###### Things done

I tested this by generating grcov coverage reports. Seems to work fine. There are a couple failing integration tests which I've disabled. These tests are looking for `grcov` in the `target/(debug|release)/grcov` and not finding them. The application itself seems to work.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/grcov/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib, rustPlatform, fetchFromGitHub, llvmPackages }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "grcov";
+  version = "v0.8.7";
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/grcov/default.nix:5]: version string should start with a digit.

File: pkgs/development/tools/misc/grcov/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    rev = version;
+    sha256 = "sha256-4McF9tLIjDCftyGI29pm/LnTUBVWG+pY5z+mGFKySQM=";
+  };
+
+  cargoSha256 = "sha256-CdsTAMiEK8udRoD9PQSC0jQbPkdynFL0Tdw5aAZUmsM=";
+
+  # tests do not find grcov path correctly
+  checkFlags = let
+    skipList = [
+    "test_coveralls_service_job_id_is_not_sufficient"
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/grcov/default.nix:19]: Indent.
Example:

```suggestion
      "test_coveralls_service_job_id_is_not_sufficient"
```

File: pkgs/development/tools/misc/grcov/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "test_integration"
+    "test_integration_guess_single_file"
+    "test_integration_zip_dir"
+    "test_integration_zip_zip"
+    ];
+  skipFlag = test: "--skip " + test;
+  in builtins.concatStringsSep " " (builtins.map skipFlag skipList);
+
+  meta = with lib; {
+    maintainers = with maintainers; [ AndersonTorres DieracDelta ];
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/grcov/default.nix:32]: I'm not so sure to handle Rust packages now. 
Do you think on maintaining this package here in Nixpkgs?

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -2975,6 +2975,12 @@
     githubId = 8404455;
     name = "Diego Lelis";
   };
+  DieracDelta = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:2978]: Put your credentials in a separate commit, with a message `maintainers: add DieracDelta`.

If you need some advice on how to do it, read this tutorials:

https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/misc/grcov/default.nix (hunk 3 above)
Comment (DieracDelta) [reply pkgs/development/tools/misc/grcov/default.nix]: Sure! Would love to.

**Comment by DieracDelta** [comment]
Thanks for the feedback! I've made some changes.

File: pkgs/development/tools/misc/grcov/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      "test_integration"
+      "test_integration_guess_single_file"
+      "test_integration_zip_dir"
+      "test_integration_zip_zip"
+    ];
+  skipFlag = test: "--skip " + test;
+  in builtins.concatStringsSep " " (builtins.map skipFlag skipList);
+
+  meta = with lib; {
+    description = "grcov collects and aggregates code coverage information for multiple source files.";
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/grcov/default.nix:32]: `description` should be
- small: do not pass 80-char limit (not a hard limit)
- not self-referent: do not use its own name (`grcov` in this example), 
- **not** end with punctuation
- descriptive

Usually you can use the summary from GitHub project page:

```suggestion
    description = "Rust tool to collect and aggregate code coverage data for multiple source files";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/misc/grcov/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{ lib, rustPlatform, fetchFromGitHub, llvmPackages }:
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/grcov/default.nix:1]: Drop unused argument

```suggestion
{ lib, rustPlatform, fetchFromGitHub }:
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/misc/grcov/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      "test_coveralls_service_job_id_is_not_sufficient"
+      "test_coveralls_service_name_is_not_sufficient"
+      "test_coveralls_works_with_just_service_name_and_job_id_args"
+      "test_coveralls_works_with_just_token_arg"
+      "test_integration"
+      "test_integration_guess_single_file"
+      "test_integration_zip_dir"
+      "test_integration_zip_zip"
+    ];
+  skipFlag = test: "--skip " + test;
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/grcov/default.nix:28]:
```suggestion
    skipFlag = test: "--skip " + test;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by DieracDelta** [comment]
I ran the grcov through `nixfmt`. It [doesn't look like there's a standardized way to format](https://discourse.nixos.org/t/formatters-contributing-guidelines/11883/8), so hopefully this is "good enough"TM. The formatting didn't change much, really (a 3 line difference).

**Comment by AndersonTorres** [comment]
Looks fine and readable to me.
